packages/router/router.ts

```diff
@@ -3655,21 +3655,27 @@ function getMatchesToLoad(
   let boundaryMatches = getLoaderMatchesUntilBoundary(matches, boundaryId);
 
   let navigationMatches = boundaryMatches.filter((match, index) => {
-    if (isInitialLoad) {
-      // On initial hydration we don't do any shouldRevalidate stuff - we just
-      // call the unhydrated loaders
-      return isUnhydratedRoute(state, match.route);
```

Comment:
This was backwards before - we should check lazy and the lack of a loader first, then look for unhydrated routes